src/browser/ui/dom/components/ReactDOMInput.js

```diff
@@ -115,6 +138,24 @@ var ReactDOMInput = ReactClass.createClass({
   _handleChange: function(event) {
     var returnValue;
     var onChange = LinkedValueUtils.getOnChange(this.props);
+
+    // IE 10+ fire input events when setting/unsetting a placeholder
+    // we guard against it by checking if the next and
+    // last values are both empty and bailing out of the change
+    // https://github.com/facebook/react/issues/3484
+    if (isIEInputEvent(event)) {
+      var controlledValue = LinkedValueUtils.getValue(this.props);
+      var lastValue = controlledValue != null ?
```

Comment:
> I want to say in those cases the "right" thing to do is use the  value  prop as the last value

That can only be applied to controlled components, not uncontrolled. Uncontrolled components are frequently updated through directly setting the `value` property. As for controlled components, don't quote me on this, but it's probably a no-go to look at the `value`-prop for controlled components, batching/deferring means that the previous value might not actually be the *last* value, only a *previous* value.